### PR TITLE
Convert X11Requirement test to spec.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

### CODE REVIEW

**Comment by reitermarkus** [comment]
@MikeMcQuaid, what is the `netlify` check used for?

**Comment by MikeMcQuaid** [comment]
@reitermarkus It's something I'm experimenting with for docs.brew.sh hosting. Going to look how to turn it off now...

File: Library/Homebrew/test/x11_requirement_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+require "requirements/x11_requirement"
+
+describe X11Requirement do
+  subject { described_class.new }
+
+  describe "#eql?" do
```
Comment (alyssais) [inline Library/Homebrew/test/x11_requirement_spec.rb]: This is testing more than just `#eql?`. Maybe `describe "equality" do`?

File: Library/Homebrew/test/x11_requirement_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+require "requirements/x11_requirement"
+
+describe X11Requirement do
+  subject { described_class.new }
```
Comment (alyssais) [inline Library/Homebrew/test/x11_requirement_spec.rb]: This line is unnecessary, I think.

File: Library/Homebrew/test/x11_requirement_spec.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    end
+  end
+
+  describe "#modify_build_environment" do
+    it "calls ENV#x11" do
+      allow(subject).to receive(:satisfied?).and_return(true)
+
+      expect(ENV).to receive(:x11)
+
+      subject.modify_build_environment
```
Comment (alyssais) [inline Library/Homebrew/test/x11_requirement_spec.rb]: There's a lot of whitespace here that could probably be condensed…

File: Library/Homebrew/test/x11_requirement_spec.rb (hunk 2 above)
Comment (reitermarkus) [inline Library/Homebrew/test/x11_requirement_spec.rb]: Thanks, didn't know `subject` was implicitly initialized.

File: Library/Homebrew/test/x11_requirement_spec.rb (hunk 1 above)
Comment (reitermarkus) [inline Library/Homebrew/test/x11_requirement_spec.rb]: Isn't this test basically checking if `eql?` is using `hash` to compare the two items? It isn't, by the way, which is odd.

Comment (alyssais) [inline Library/Homebrew/test/x11_requirement_spec.rb]: If that's what it's checking, it should say that in the description!

Comment (reitermarkus) [inline Library/Homebrew/test/x11_requirement_spec.rb]: Well, I don't know what was originally intended to be checked, but that's what I'm guessing should be checked.

File: Library/Homebrew/requirement.rb [hunk 4]
```diff
@@ -109,7 +109,7 @@ def env_proc
   end
 
   def ==(other)
-    instance_of?(other.class) && name == other.name && tags == other.tags
+    instance_of?(other.class) && hash == other.hash
```
Comment (alyssais) [inline Library/Homebrew/requirement.rb]: What's the reason for this change?

Comment (reitermarkus) [inline Library/Homebrew/requirement.rb]: The test is now expecting `eql?` to call `hash`, which it wasn't before. Anyway, I think it would be better to move the `hash` check to a separate test, so ignore this.

**Comment by reitermarkus** [comment]
I removed the `hash` equality check from this test and added a separate test for it to `requirement_spec.rb`.
